backend/src/test/java/com/redhat/cloud/notifications/routers/EmailsOnlyModeTest.java

```diff
@@ -0,0 +1,170 @@
+package com.redhat.cloud.notifications.routers;
+
+import com.redhat.cloud.notifications.Json;
+import com.redhat.cloud.notifications.MockServerConfig;
+import com.redhat.cloud.notifications.TestHelpers;
+import com.redhat.cloud.notifications.TestLifecycleManager;
+import com.redhat.cloud.notifications.config.FeatureFlipper;
+import com.redhat.cloud.notifications.db.DbIsolatedTest;
+import com.redhat.cloud.notifications.db.repositories.EndpointRepository;
+import com.redhat.cloud.notifications.models.Endpoint;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.InjectMock;
+import io.restassured.http.Header;
+import org.junit.jupiter.api.Test;
+
+import javax.inject.Inject;
+import java.util.UUID;
+
+import static com.redhat.cloud.notifications.Constants.API_INTEGRATIONS_V_1_0;
+import static com.redhat.cloud.notifications.MockServerConfig.RbacAccess.FULL_ACCESS;
+import static com.redhat.cloud.notifications.models.EndpointType.CAMEL;
+import static com.redhat.cloud.notifications.models.EndpointType.WEBHOOK;
+import static com.redhat.cloud.notifications.routers.EndpointResource.UNSUPPORTED_ENDPOINT_TYPE;
+import static io.restassured.RestAssured.given;
+import static io.restassured.http.ContentType.JSON;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+
+@QuarkusTest
+@QuarkusTestResource(TestLifecycleManager.class)
+public class EmailsOnlyModeTest extends DbIsolatedTest {
+
+    @Inject
+    FeatureFlipper featureFlipper;
+
+    @InjectMock
+    EndpointRepository endpointRepository;
+
+    @Test
+    void testCreateUnsupportedEndpointType() {
+        featureFlipper.setEmailsOnlyMode(true);
+
+        String identityHeaderValue = TestHelpers.encodeRHIdentityInfo("account-id", "org-id", "username");
+        Header identityHeader = TestHelpers.createRHIdentityHeader(identityHeaderValue);
+        MockServerConfig.addMockRbacAccess(identityHeaderValue, FULL_ACCESS);
+
+        Endpoint endpoint = new Endpoint();
+        endpoint.setType(WEBHOOK);
+        endpoint.setName("name");
+        endpoint.setDescription("description");
+
+        String responseBody = given()
+                .basePath(API_INTEGRATIONS_V_1_0)
+                .header(identityHeader)
+                .when()
+                .contentType(JSON)
+                .body(Json.encode(endpoint))
+                .post("/endpoints")
+                .then()
+                .statusCode(400)
+                .extract().asString();
+        assertEquals(UNSUPPORTED_ENDPOINT_TYPE, responseBody);
```

Comment:
if this assert fails, it won't process the `featureFlipper.setEmailsOnlyMode(false);` which might break more tests, making it harder to track it down to the real error.
Could you wrap this into a `try {} finally { featureFlipper.setEmailsOnlyMode(false); }`  or have a `afterEach` setting the mode to `false`?

This applies for all the other tests.

Reply:
Good catch, fixed!